ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -993,6 +1011,174 @@ const triggerSet: TriggerSet<Data> = {
+          // 16 = short, 20 = long
+          data.pangenesisRole[matches.target] = duration > 18 ? 'longDark' : 'shortDark';
+        } else if (id === pangenesisEffects.lightTilt) {
+          const duration = parseFloat(matches.duration);
+          // 16 = short, 20 = long
+          data.pangenesisRole[matches.target] = duration > 18 ? 'longLight' : 'shortLight';
+        } else if (id === pangenesisEffects.unstableFactor) {
+          if (matches.count === '01')
+            data.pangenesisRole[matches.target] = 'one';
+        } else if (id === pangenesisEffects.stableSystem) {
+          // Ordered: Unstable Factor / Stable System / Umbral Tilt (light) / Astral Tilt (dark)
+          // ...and applied per person in that order.  Don't overwrite roles here.
+          data.pangenesisRole[matches.target] ??= 'not';
+        }
+      },
+    },
+    {
+      id: 'P12S Pangenesis Initial',
+      type: 'GainsEffect',
+      netRegex: { effectId: 'E22', capture: false },
+      delaySeconds: 0.5,
+      durationSeconds: (data) => {
+        // There's ~13 seconds until the first tower and ~18 until the second tower.
+        const myRole = data.pangenesisRole[data.me];
+        return myRole === 'not' || myRole === 'longDark' || myRole === 'longLight' ? 17 : 12;
+      },
+      suppressSeconds: 999999,
+      alertText: (data, _matches, output) => {
+        const myRole = data.pangenesisRole[data.me];
+        if (myRole === undefined)
+          return;
+
+        if (myRole === 'shortLight')
+          return output.shortLight!();
+        if (myRole === 'longLight')
+          return output.longLight!();
+        if (myRole === 'shortDark')
+          return output.shortDark!();
+        if (myRole === 'longDark')
+          return output.longDark!();
+
+        const myBuddy = Object.keys(data.pangenesisRole).find((x) => {
+          return data.pangenesisRole[x] === myRole && x !== data.me;
+        });
+        const player = myBuddy === undefined ? output.unknown!() : data.ShortName(myBuddy);
+        if (myRole === 'not')
+          return output.nothing!({ player: player });
+        return output.one!({ player: player });
+      },
+      run: (data) => data.pangenesisDebuffsCalled = true,
+      outputStrings: {
+        nothing: {
+          en: 'Nothing (w/${player})',
+        },
+        one: {
+          en: 'One (w/${player})',
+        },
+        shortLight: {
+          en: 'Short Light (get first dark)',
+        },
+        longLight: {
+          en: 'Long Light (get second dark)',
+        },
+        shortDark: {
+          en: 'Short Dark (get first light)',
+        },
+        longDark: {
+          en: 'Long Dark (get second light)',
+        },
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      id: 'P12S Pangenesis Tilt Gain',
+      type: 'GainsEffect',
+      netRegex: { effectId: [pangenesisEffects.lightTilt, pangenesisEffects.darkTilt] },
+      condition: Conditions.targetIsYou(),
```

Comment:
```suggestion
      condition: (data) => Conditions.targetIsYou() && !data.isDoorBoss,
```

Reply:
(Mmm you can't do this, since targetIsYou() returns a function.  I think I got all of them in the PR I submitted.  Thanks for pointing that out.)